protocols/Convert_Final_report_to_Plink.sh

```diff
@@ -0,0 +1,160 @@
+grep -P '^[123456789]' "${tmpPlinkDir}/${arrayTmpMap}" | sort -k1n -k4n > "${tmpPlinkDir}/${arrayMapFile}"
+grep -P '^[X]\s' "${tmpPlinkDir}/${arrayTmpMap}" | sort -k4n >> "${tmpPlinkDir}/${arrayMapFile}"
+grep -P '^[Y]\s' "${tmpPlinkDir}/${arrayTmpMap}" | sort -k4n >> "${tmpPlinkDir}/${arrayMapFile}"
+
+#####################################
+##Create .bed and other files (keep sample from sample_list.txt).
+##Create .bed and other files (keep sample from sample_list.txt).
+
+
+#Create ped and map file
+plink \
+--lfile "${tmpPlinkDir}/${Sample_ID}" \
+--recode \
+--noweb \
+--out "${tmpPlinkDir}/${Sample_ID}" \
+--keep "${tmpPlinkDir}/${familyList}"
+
+#Convert ped and map files to a VCF file
+
+#Use different version from plink to make the VCF file
+module unload plink
+module load "${PLINKVersion2}"
+module list
+
+##Create genotype VCF for sample
+	plink \
+	--recode vcf-iid \
+	--ped "${tmpPlinkDir}/${Sample_ID}.ped" \
+	--map "${tmpPlinkDir}/${arrayMapFile}" \
+	--out "${tmpPlinkDir}/${Sample_ID}"
+
+
+# Filter VCF file with bed file of SNPs from HRC with a MAF value lower than 0.01 .
+# we want to exclude those SNP's so we use the -v option from bedtools intersect
+
+
+module load "${BEDtoolsVersion}"
+module load "${HTSlibVersion}"
+module list
+
+bedtools intersect -a "${tmpPlinkDir}/${Sample_ID}.vcf" -b "${HRCFilterBedFile}" -v -header >  ${tmpPlinkDir}/${Sample_ID}.filteredMAF.vcf
+
+bgzip -c "${tmpPlinkDir}/${Sample_ID}.filteredMAF.vcf" > "${tmpPlinkDir}/${Sample_ID}.filteredMAF.vcf.gz"
+tabix -p vcf "${tmpPlinkDir}/${Sample_ID}.filteredMAF.vcf.gz"
+
+# Make an VCF per chromosome
+
+chromsomes=()
+
+for chromosome in "${chr[@]}"
+do
+	array_contains chromosomes "${chromosome}" || chromosomes+=("$chromosome")
+done
+
+for chr in ${chromosomes[@]}
+do
+	tabix -h "${tmpPlinkDir}/${Sample_ID}.filteredMAF.vcf.gz" "${chr}" > "${tmpPlinkDir}/chr${chr}_${Sample_ID}.filteredMAF.vcf"
+done
+
+#Remove duplicate SNP's from the VCF files
+
+module load "${BCFtoolsVersion}"
+
+for chr in ${chromosomes[@]}
+do
+bcftools norm -d any "${tmpPlinkDir}/chr${chr}_${Sample_ID}.filteredMAF.vcf" -O v -o "${tmpPlinkDir}/chr${chr}_${Sample_ID}.filteredMAF_duplicatesRemoved.vcf"
+done
+
+
+# Convert per chromosome VCF's to .bed , .bim , .fam format PLINK which can be used as phasing input
+
+for chr in ${chromosomes[@]}
+do
+	plink \
+        --vcf "${tmpPlinkDir}/chr${chr}_${Sample_ID}.filteredMAF_duplicatesRemoved.vcf" \
+        --make-bed \
+        --out "${tmpPlinkDir}/chr${chr}_${Sample_ID}"
+done
+
+
+# Move output to output folder
+
+for chr in ${chromosomes[@]}
+do
+	echo "mv temporaru results from ${tmpPlinkDir} to ${PlinkDir}"
+	mv "${tmpPlinkDir}/chr${chr}_${Sample_ID}.filteredMAF_duplicatesRemoved.vcf" "${PlinkDir}"
+	mv "${tmpPlinkDir}/chr${chr}_${Sample_ID}.bed" "${PlinkDir}"
+	mv "${tmpPlinkDir}/chr${chr}_${Sample_ID}.bim" "${PlinkDir}"
+	mv "${tmpPlinkDir}/chr${chr}_${Sample_ID}.fam" "${PlinkDir}"
+
+done
```

Comment:
Je doet vijf keer een loop over chromosomes kan je dat niet in een loop doen? of heb je hier bewust voor gekozen?